### DISCUSSION TITLE
Create a specific mask (no train nor test)

### DISCUSSION THREAD

**Discussion by AlejandroTL** [pyg-team/pytorch_geometric#5081] [Q&A] [answered]
Hello!

First of all: this package is simply amazing, thanks.

My doubt is how can I create a specific mask for the huge graph that I have. In the pipeline I'm currently using I have a huge graph but I want to input, at different parts of the model, two subgraphs of the graph. 
The first imputation is the huge graph with the exception of a few nodes, the second part is a much smaller subgraph that involves the previously ignored nodes. I assume that the best way of doing it is using a mask (or two masks, actually) over the huge graph that I have already build. I don't want these masks to be train masks or whatever, is just the way in which I think its easier to generate the subgraphs that I want, instead of generating them from scratch given that they share same topology and features.

Thanks!

**Comment by wsad1** [comment, marked as answer]
[subgraph](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/utils/subgraph.html) can you call this twice, with different masks. It creates a new `edge_index` so you don't have to create features from scratch.
